WeCantSpell.Hunspell/Infrastructure/MemoryEx.cs

```diff
@@ -17,10 +17,35 @@ public static int IndexOf(this ReadOnlySpan<char> @this, ReadOnlySpan<char> valu
         return result >= 0 ? result + startIndex : result;
     }
 
-    public static int IndexOf(this Span<char> @this, ReadOnlySpan<char> value, int startIndex)
+    public static bool SortedContains(ReadOnlySpan<char> sorted, char value)
     {
-        var result = @this.Slice(startIndex).IndexOf(value);
-        return result >= 0 ? result + startIndex : result;
+        return sorted.Length switch
+        {
+            0 => false,
+            1 => sorted[0] == value,
+            2 => sorted[0] == value || sorted[1] == value,
+            3 => sorted[0] == value || sorted[1] == value || sorted[2] == value,
+            <= 8 => checkIterative(sorted, value),
```

Comment:
Is `8` the right magic number?

Reply:
Everything seems to go the ICollection<T> route.

Reply:
It's really hard to know for sure, but I'm thinking a nudge lower would be better.